src/smif/app/src/components/ConfigForm/SosModel/DependencyList.js

```diff
@@ -69,13 +69,28 @@ class DependencyList extends Component {
     handleSubmit(event) {
         event.preventDefault()
 
-        if (this.state.formEditMode) {
-            this.props.dependencies[this.state.formEditNumber] = this.state.dependency
+        // Prevent user from adding existing dependencies
+        if (this.props.dependencies.filter((dependency, idx) => {
+            return (
+                dependency.source == this.state.dependency.source &&
+                dependency.source_output == this.state.dependency.source_output &&
+                dependency.sink == this.state.dependency.sink &&
+                dependency.sink_input == this.state.dependency.sink_input &&
+                idx != parseInt(this.state.formEditNumber)
+            )
+        }).length > 0) {
+            alert('Cannot save dependency because it is already existing')
```

Comment:
Change this message to:

`'Cannot save dependency because it already exists'`

Reply:
Done